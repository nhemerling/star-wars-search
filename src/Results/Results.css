.Results {
    padding: 20px 50px;
    margin: 30px;
}

ul {
    list-style: none;
    text-align: left;
    margin: 0;
    padding: 0;
}

h2 {
    text-align: left;
}

