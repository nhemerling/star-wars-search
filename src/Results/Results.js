import React from 'react';
import './Results.css';

export default function Results(props) {
    return (
        <div className="Results">
            {/* map over the results and return the names of the characters with some details */}
            {props.results.map(result => 
                <div className={result.name} key={result.name}>
                    {props.type !== 'films' && <h2>{result.name}</h2>}
                    {props.type === 'films' && <h2>{result.title}</h2>}
                    {props.type === 'people' &&
                        <ul>
                            <li>Height: {result.height} cm</li>
                            <li>Mass: {result.mass} kg</li>
                            <li>Birth Year: {result.birth_year}</li>
                        </ul>
                    }
                    {props.type === 'starships' &&
                        <ul>
                            <li>Hyperdrive Rating: {result.hyperdrive_rating}</li>
                            <li>Manufacturer: {result.manufacturer}</li>
                            <li>Starship Class: {result.starship_class}</li>
                        </ul>
                    }
                    {props.type === 'vehicles' &&
                        <ul>
                            <li>Length: {result.length} m</li>
                            <li>Manufacturer: {result.manufacturer}</li>
                            <li>Model: {result.model}</li>
                        </ul>
                    }
                    {props.type === 'species' &&
                        <ul>
                            <li>Average Height: {result.average_height} m</li>
                            <li>Designation: {result.designation}</li>
                            <li>Language: {result.language}</li>
                        </ul>
                    }
                    {props.type === 'planets' &&
                        <ul>
                            <li>Climate: {result.climate}</li>
                            <li>Population: {result.population}</li>
                            <li>Terrain: {result.terrain}</li>
                        </ul>
                    }
                    {props.type === 'films' &&
                        <ul>
                            <li>Episode: {result.episode_id}</li>
                            <li>Director: {result.director}</li>
                            <li>Opening Crawl: {result.opening_crawl}</li>
                        </ul>
                    }
                </div>
            )}
        </div>
    );
}

Results.defaultProps = {
    results: [],
    type: 'people'
}