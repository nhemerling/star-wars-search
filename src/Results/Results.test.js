import React from 'react';
import ReactDOM from 'react-dom';
import Results from './Results';
import renderer from 'react-test-renderer';

describe('Results component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Results />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected with empty results array', () => {
    const tree = renderer
      .create(<Results
                results={[]} type="people" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the UI as expected with empty results array and type set as vehicles', () => {
    const tree = renderer
      .create(<Results
                results={[]} type="vehicles" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the UI as expected with no props', () => {
    const tree = renderer
      .create(<Results />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
