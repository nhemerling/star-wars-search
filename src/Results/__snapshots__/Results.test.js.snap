// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results component renders the UI as expected with empty results array 1`] = `
<div
  className="Results"
/>
`;

exports[`Results component renders the UI as expected with empty results array and type set as vehicles 1`] = `
<div
  className="Results"
/>
`;

exports[`Results component renders the UI as expected with no props 1`] = `
<div
  className="Results"
/>
`;
